File: iut/TP7_TP8/conjugaison.py
def conjugaison(verbe):
    voy = "aeiouyAEIOUY"

    
    i = len(verbe)

    
    if i >= 2 and verbe[-1] == 'r' and verbe[-2] == 'e':
        
        radical = verbe[:-2]

        
        temp = 0
        for v in voy:
            if radical[0] == v:
                temp = 1
                break

        
        if temp == 1:
            print(f"j'{radical}e")
        else:
            print(f"je {radical}e")

        print(f"tu {radical}es")
        print(f"il {radical}e")
        print(f"nous {radical}ons")
        print(f"vous {radical}ez")
        print(f"elles {radical}ent")

    else:
        print("Erreur : le mot ne se termine pas par 'er'.")



mot = "ester"
conjugaison(mot)

File: iut/TP7_TP8/conjugaison.c
#include <stdio.h>
#include <stdlib.h>

void conjugaison(char verbe[])
{
    int i,l;
    char *voy = "aeiouyAEIOUY";
    int temp;

    i = 0;
    l = 0;
    temp = 0;
    while (verbe[i])
    {   
        i++;
    }

    if (verbe[i-1] == 'r' && verbe[i-2] == 'e')
    {
        char *test;
        test = (char *) malloc(i-2 * sizeof(char));
        test[i] = '\0';


        while (l != i-2)
        {
            test[l] = verbe[l];
            l++;
        }
        l = 0;
        

        while (voy[l])
        {
            if (voy[l] == test[0])
            {
                temp =1;
                break;
            }
            l++;
        }
        
        if (temp == 1)
            printf("j'%se\n", test);
        else
            printf("je %se\n", test);
        
        
        printf("tu %ses\n", test);
        printf("il %se\n", test);
        printf("nous %sons\n", test);
        printf("vous %sez\n", test);
        printf("elles %sent\n", test);

        free(test);
    }
    else
    {
        printf("Erreur");
    }
    
    
}

int main(void)
{
    char mot[] = "ester";
    conjugaison(mot);
    
    return 0;
}
